pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -111,6 +112,12 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		}
 	}
 
+	// If the StatefulSet is being deleted, don't do anything other than updating
+	// status.
+	if set.DeletionTimestamp != nil {
+		return nil
```

Comment:
@kow3ns we should have a helper that would do just that - iterate over all pods and update the status accordingly.